client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/channel/failover/strategies/DefaultFailoverStrategy.java

```diff
@@ -73,6 +73,9 @@ public FailoverDecision onFailover(FailoverStatus failoverStatus) {
                 return new FailoverDecision(FailoverAction.RETRY, operationsServersRetryPeriod, timeUnit);
             case NO_CONNECTIVITY:
                 return new FailoverDecision(FailoverAction.RETRY, noConnectivityRetryPeriod, timeUnit);
+            case ENDPOINT_VERIFICATION_FAILED:
+            case CREDENTIALS_REVOKED:
+                return new FailoverDecision(FailoverAction.STOP_APP);
```

Comment:
STOP_APP failover action is invalid. SDK should not control APP behaviour. 
However, STOP_SDK is ok. There is a listener in KaaClient that should be notified on KaaClient stop. 
We need to replace STOP_APP action with STOP_SDK and add corresponding changes. @abohomol please create ticket for this. In this case let's do a retry instead of System.exit(0);